lib/3scale/backend/errors.rb

```diff
@@ -67,6 +67,12 @@ def initialize(id, user_key)
       end
     end
 
+    class ApplicationHasNoState < Error
```

Comment:
I would let exception middleware catche the error and set http status code on Error type. So, I would inherit from BadRequest error class.